### ISSUE TITLE
Thunderbird VEVENT with the greater SEQUENCE is not given in event.instance.vevent

### ISSUE THREAD

**Issue by DirksProg** [python-caldav/caldav#32]
From Thunderbird Lighting I got events with two VEVENT-Parts in it. The first part is with a "X-MOZ-FAKED-MASTER" which have no summary and other missing stuff. Directly with the same UID there is another vevent with the right stuff inside and a SEQUENCE. But with caldav I just get the first broken VEVENT. How can I get the one with the highest SEQUENCE, or maybe just both, I can choose the right SEQUENCE by myself? As myevent is not iterable, I have no chance to see the second part of this ics.

Sample Code:
```
import caldav
events = calendar.date_search(start, end)
for event in events :
                myevent = event.instance.vevent
```

Here the ics-File I get the wrong SEQUENCE:

```
BEGIN:VCALENDAR
PRODID:-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN
VERSION:2.0
BEGIN:VTIMEZONE
TZID:Europe/Berlin
BEGIN:DAYLIGHT
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
TZNAME:CEST
DTSTART:19700329T020000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3
END:DAYLIGHT
BEGIN:STANDARD
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
TZNAME:CET
DTSTART:19701025T030000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
CREATED:20180315T201819Z
LAST-MODIFIED:20180315T201840Z
DTSTAMP:20180315T201840Z
UID:7c7c51c4-d0ae-4837-b7d9-f3379292314c
RDATE;VALUE=DATE-TIME:20180830T140000Z
DTSTART;TZID=Europe/Berlin:20180830T160000
DTEND;TZID=Europe/Berlin:20180830T160000
X-MOZ-FAKED-MASTER:1
X-MOZ-GENERATION:1
END:VEVENT
BEGIN:VEVENT
CREATED:20180315T201802Z
LAST-MODIFIED:20180315T201840Z
DTSTAMP:20180315T201840Z
UID:7c7c51c4-d0ae-4837-b7d9-f3379292314c
SUMMARY:This is the true date
RECURRENCE-ID;TZID=Europe/Berlin:20180830T160000
DTSTART;TZID=Europe/Berlin:20180830T160000
DTEND;TZID=Europe/Berlin:20180830T180000
DESCRIPTION:My Date\n
SEQUENCE:1
END:VEVENT
END:VCALENDAR
```

A workaroung might be giving me all VEVENTs, and I decide which one to use.

**Comment by tobixen** [comment]
I think that this may be a server-side issue, not a client side issue.

It's correct that when the SEQUENCE number is given in an icalendar object, the object with the biggest SEQUENCE number is to be considered the correct one.  However, I believe the server should handle this logic and serve the latest object.

What server do you use?

Perhaps we need to write some test code for reproducing this issue

**Comment by DirksProg** [comment]
I think it was Nextcloud 13.0.2. 
Nextcloud send the whole ICS as you see above. The ICS is the raw version from the server.

Example code:
```
import caldav
events = calendar.date_search(start, end)
for event in events :
                myevent = event.instance.vevent
```

myevent got only the part with
```
RDATE;VALUE=DATE-TIME:20180830T140000Z
DTSTART;TZID=Europe/Berlin:20180830T160000
DTEND;TZID=Europe/Berlin:20180830T160000
X-MOZ-FAKED-MASTER:1
X-MOZ-GENERATION:1
```
and so on,

```
BEGIN:VEVENT
CREATED:20180315T201802Z
LAST-MODIFIED:20180315T201840Z
DTSTAMP:20180315T201840Z
UID:7c7c51c4-d0ae-4837-b7d9-f3379292314c
SUMMARY:This is the true date
RECURRENCE-ID;TZID=Europe/Berlin:20180830T160000
DTSTART;TZID=Europe/Berlin:20180830T160000
DTEND;TZID=Europe/Berlin:20180830T180000
DESCRIPTION:My Date\n
SEQUENCE:1
END:VEVENT
```
is lost. :(

**Comment by tobixen** [comment]
I should set up test servers with nextcloud (or see if I can use some existing servers) and see if I can run the tests towards it.  We also need test code to reproduce the issue.

**Comment by tobixen** [comment]
Actually, this is probably a "bug" in vobjects and will probably be solved in 1.0 as I plan to migrate to icalendar.  I was encountering the same thing when looking into recurring events, ref #86.  The "expand"-flag to date_search works on most server implementations, but the expanded stuff is lost when doing event.instance.vevent.  It's still possible to fish it out by using event.instance.components()

I'll close this.
